yoyodyne/models/transducer.py

```diff
@@ -154,7 +154,7 @@ def decode(
                 break
             # Proceeds to make new edit; new action for all current decoding.
             action_count = torch.where(
-                not_complete, action_count + 1, action_count
+                not_complete.to(self.device), action_count + 1, action_count
```

Comment:
`not_complete` should already be on device no? If not, this should be managed by its instantiation.

Reply:
I don't know how. Can you suggest an alternative to line 152:

`not_complete = last_action != self.actions.end_idx`

that creates `not_complete` on `self.device` rather than what I've done here? This is where I don't know enough torch (or numpy really) to suggest something better.

Reply:
This should automatically cast to the same device as `last_action` since torch tries to share devices when possible. Was this not the case when you tested?

Reply:
I believe this was the thing that was giving me the error with heterogeneous devices, yes. Does `last_action` not itself have a consistent device?

Reply:
Was just checking the code. It's explicitly cast to a device each part of the decode step. Leave your change as is and I'll look into it in a separate PR. That's would be a bug.

Reply:
Roger that. Separate PR could trace this down or remove .to if I'm wrong and it's not necessary anymore.